### PR TITLE
fix(world): register Delegations table in CoreModule

### PR DESCRIPTION
**Pull request by alvrs**
small patch to register the Delegations table so it can be synced by the sync stack

### CODE REVIEW

File: .changeset/quiet-dancers-prove.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@latticexyz/world": patch
+---
+
+- Register `Delegations` table in the `CoreModule`
```
Comment (holic) [inline .changeset/quiet-dancers-prove.md:5]: do we need the bullet here?

File: packages/world/src/modules/core/CoreModule.sol [hunk 2]
```diff
@@ -58,6 +59,7 @@ contract CoreModule is IModule, WorldContextConsumer {
    */
   function _registerCoreTables() internal {
     InstalledModules.register();
+    Delegations.register();
```
Comment (holic) [inline packages/world/src/modules/core/CoreModule.sol:62]: any easy way to add some tests to ensure we're registering all of our expected core tables?

**Review by holic — DISMISSED** [review]

File: .changeset/quiet-dancers-prove.md (hunk 1 above)
Comment (alvrs) [inline .changeset/quiet-dancers-prove.md:5]:
```suggestion
Register `Delegations` table in the `CoreModule`
```

File: packages/world/src/modules/core/CoreModule.sol (hunk 2 above)
Comment (alvrs) [reply packages/world/src/modules/core/CoreModule.sol:62]: no automated way, maintaining the test would have the same issue as maintaining the actual registration

Comment (alvrs) [reply packages/world/src/modules/core/CoreModule.sol:62]: well we could add some sort of e2e test that checks that all tables in the world config are registered, but for that we first need to separate the test tables

Comment (holic) [reply packages/world/src/modules/core/CoreModule.sol:62]: opened an issue to track here: https://github.com/latticexyz/mud/issues/1456

**Pull request merged**